addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -148,6 +148,8 @@ return AbstractModel.extend({
         if (f) {
             if (f.active !== filter.active) {
                 f.active = filter.active;
+            } else if (filter.toRemove) {
```

Comment:
@SimonGenin I didn't get your statement, the ```else if``` is added because if you first inactivate filter from calendar sidebar and then later you try to delete then first ```if``` statement will not get true so it will go to else and retrun false, do you mean we don't need ```else``` at all?